orchestra/static/orchestra/task/js/services.js

```diff
@@ -47,9 +47,26 @@ serviceModule.factory('orchestraService', function() {
         }
     }
 
+    var signals = {
+        // Functions to override in specific task controllers.
+        'save': {
```

Comment:
high-level: we don't want callers to override this, because we might want two callers to subscribe to signals

the way this can happen is:
- we don't expose `signals`---that stays internal
- we expose a function `registerSignal(signalType, state, callback)` that does `signals[signalType][signal].append(callback)` (Python syntax, I know :))
- we have an internal function called `fireSignal(signalType, state)` that loops through `signals[signalType][signal]` and calls the callbacks.
- replace all of the calls above, e.g., `orchestraService.signals.save.finally();` -> `fireSignal(signalType, state);`
- i'm not even sure we need to separate `signalType` and `state`: you could imagine having a flatter structure, and we can name signals things like `submit.before`